website_sale_contract/wizards/contract_checkout.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ContractCheckout(models.TransientModel):
+
+    _name = 'contract.checkout'
+    _description = 'Contract Checkout'
+
+    order_id = fields.Many2one(
+        string='Order',
+        comodel_name='sale.order',
+        default=lambda s: s._default_order_id(),
+        required=True,
+    )
+    contract_ids = fields.One2many(
+        string='Contracts',
+        comodel_name='contract.temp',
+        inverse_name='contract_checkout_id',
+    )
+
+    @api.model
+    def _default_order_id(self):
+        if self.env.context.get('active_model') != 'sale.order':
+            return
+        return self.env['sale.order'].browse(
+            self.env.context.get('active_id')
+        )
+
+    @api.multi
+    def _create_temp_lines(self, sale_order_lines):
+        self.ensure_one()
+        ContractTemp = self.env['contract.temp']
```

Comment:
Yeah exactly about the linked list.

You mentioned in the #65 that the contracts won't be created until the final payment process has been completed, so in order to ensure those contracts have the signatory field data saved, there needed to be somewhere to temporarily save them.